### PR TITLE
feature 124 Ui Components Month

### PR DESCRIPTION
**Pull request by DanielHodgeGH**
PR to add:

NOX-225 View LatLong
NOX-228 View StreetAddress
NOX-229 View entity
NOX-124 Month
Nox-109 DayOfWeek

Also renamed UiEditEntitySelectLandLord_g to UiEditEntityExactlyOneLandLord_g to make the entity relationship specific

@anna-naumova25 - just checking did you mean to comment out a few Ui component fields on the VendingMachine ADD & EDIT forms? I have left it as is for now thanks

### CODE REVIEW

**Comment by anna-naumova25** [comment]
@DanielHodgeGH  please see my comment in the following PR - https://github.com/NoxOrg/Nox.Generator/pull/1118. We can discuss it on call if you need more details.

File: samples/Cryptocash.Ui/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Component/UiViewStreetAddressBase.g.cs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+        #endregion
+
+         public string DisplayStreetAddress
+        {
+            get
+            {
+                string RtnStreetAddress = String.Empty;
+
+                if (StreetAddress != null)
```
Comment (anna-naumova25) [inline samples/Cryptocash.Ui/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Component/UiViewStreetAddressBase.g.cs]: I'd replace it to 
```
if (StreetAddress == null)
    return RtnStreetAddress;
```
to remove at least one nesting level

File: samples/Cryptocash.Ui/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Component/UiViewStreetAddressBase.g.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+        [Parameter]
+        public StreetAddressDto? StreetAddress { get; set; }
+
+        [Parameter]
+        public List<CountryDto>? CountrySelectionList { get; set; }
+
+        #endregion
+
+         public string DisplayStreetAddress
```
Comment (anna-naumova25) [inline samples/Cryptocash.Ui/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Component/UiViewStreetAddressBase.g.cs:22]: I would refactor this property in the following way:
1. Define a list with strings to be included to RtnStreetAddress -  `var parts = new List<string>();`
2. Create a function to add non empty strings to the list:
```
private void AddNonEmpty(string? value, List<string> parts)
{
    if (!string.IsNullOrWhiteSpace(value))
    {
        parts.Add(value.Trim());
    }
}
```
3. Replace all the checks to `AddNonEmpty(StreetAddress.StreetNumber, parts);`
4. Join the list with comma - `RtnStreetAddress = string.Join(",  ", parts).Trim(); `

Feel free to use other names for variables/functions, it's just an example.

**Review by anna-naumova25 — CHANGES_REQUESTED** [review]
looks good, only a few comments

File: samples/Cryptocash.Ui/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Component/UiViewStreetAddressBase.g.cs (hunk 2 above)
Comment (DanielHodgeGH) [reply samples/Cryptocash.Ui/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Component/UiViewStreetAddressBase.g.cs:22]: thanks @anna-naumova25  - that is a lot better - have made change in latest commit

**Review by anna-naumova25 — APPROVED** [review]

**Review by rochar — APPROVED** [review]

**Pull request merged**